validator/client/grpc-api/grpc_validator_client.go

```diff
@@ -143,13 +147,80 @@ func (grpcValidatorClient) GetAggregatedSelections(context.Context, []iface.Beac
 }
 
 func NewGrpcValidatorClient(cc grpc.ClientConnInterface) iface.ValidatorClient {
-	return &grpcValidatorClient{ethpb.NewBeaconNodeValidatorClient(cc)}
-}
-
-func (c *grpcValidatorClient) StartEventStream(context.Context) error {
-	panic("function not supported for gRPC client")
+	return &grpcValidatorClient{ethpb.NewBeaconNodeValidatorClient(cc), false}
+}
+
+func (c *grpcValidatorClient) StartEventStream(ctx context.Context, topics []string, eventsChannel chan<- *eventClient.Event) {
+	ctx, span := trace.StartSpan(ctx, "validator.gRPCClient.StartEventStream")
+	defer span.End()
+	if len(topics) == 0 {
+		eventsChannel <- &eventClient.Event{
+			EventType: eventClient.EventError,
+			Data:      []byte(errors.New("no topics were added").Error()),
+		}
+		return
+	}
+	// TODO: ONLY WORKS WITH HEAD TOPIC RIGHT NOW/ONLY PROVIDES THE SLOT
+	containsHead := false
+	for i := range topics {
+		if topics[i] == eventClient.EventHead {
+			containsHead = true
+		}
+	}
+	if !containsHead {
+		log.Warnf("gRPC only supports the head topic, topics provided are %v.. continuing stream with head topic", topics)
```

Comment:
Why would we listen to head when the caller is not interested in it? It won't help with anything. I would return an error event in such case. A warning makes sense when there is more than 1 topic including head. In that case we should inform that other topics will be ignored.